Tests/scripts/collect_tests/collect_tests.py

```diff
@@ -871,6 +920,14 @@ def _collect_single(self, path: Path) -> Optional[CollectionResult]:
                     pack_id=pack_id, reason_description=reason_description,
                     changed_file_path=path, content_item_range=content_item_range
                 )
+
+            if file_type in XSIAM_COMPONENT_FILES:
```

Comment:
can we combine it with the previous condition (modeling rule), and call a function that separates the cases? it'll be easier to follow and debug

Reply:
Not sure that this is what you had in mind, tell me if I got it wrong.

Reply:
I meant to have only one method for collecting xsiam content (and modeling rules are always such), and in it, have `if type is modeling rules, <then the logic of the existing flow>, else <the logic you added here>`, reducing the number of functions.

Reply:
the problem with this is that the modeling rules is being collected through that function for nightly as well. Because our nightly accidentally works fine because of this, I don't want to be testing nightly flows for this issue. (Since on-push behaviour should be fixed asap)

But it can be handled in the following issue for fixing the nightly.
https://jira-hq.paloaltonetworks.local/browse/CIAC-5488

I'll add your comment there.